cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -501,6 +501,16 @@ def set_final_status():
             msg = 'Stopped services: {}'.format(','.join(failing_services))
             hookenv.status_set('blocked', msg)
             return
+    else:
+        # if we don't have components starting, we're waiting for that and
+        # shouldn't fall through to Kubernetes master running.
+        if (is_state('cni.available')):
+            hookenv.status_set('blocked',
+                               'Waiting for master components to start')
+        else:
+            hookenv.status_set('blocked',
+                               'Waiting for CNI plugins to become available')
+        return
```

Comment:
This `else` block is now handling the case when `kubernetes-master.components.started` is **not** set.  Consider getting rid of the now unnecessary `components_started` calls below.

Maybe add a comment around here that says "From this point forward, we know master components are started; let's see if we're waiting or active".

Reply:
To be clear you just want clarification in a comment? Are you implying that this masks possible messages? I didn't and don't see that, but it is possible. The components_started check just verifies that components are started and addons are not configured before it complains, so we won't go into the else. It is unnecessary to check component_started at that point though as we know that they will be started. Is that the desire? I pull that since we know it is true?

Reply:
> To be clear you just want clarification in a comment?

Yup, just reminding future people that if they tuck another status message in below this, we will have already handled `master.components.started`.  I'm trying to prevent the next person from saying `if not components_started` down below, since that would never be true once you pass this if/else.

> It is unnecessary to check component_started below the else I am adding though as we know that they will be started. Is that the desire? I pull that since we know it is true?

Bingo. Pull out the 2 references to `components_started` since they don't do anything anymore, and may just confuse the next person as to why we're checking something that will always be true.